rln/src/protocol.rs

```diff
@@ -103,121 +171,90 @@ pub enum ProofError {
     SynthesisError(#[from] SynthesisError),
 }
 
-// XXX This is different from zk-kit API:
-// const witness = RLN.genWitness(secretHash, merkleProof, epoch, signal, rlnIdentifier)
-// const fullProof = await RLN.genProof(witness, wasmFilePath, finalZkeyPath)
-//
-
-// TODO Change API here
-/// Generates a semaphore proof
+/// Generates a RLN proof
 ///
 /// # Errors
 ///
 /// Returns a [`ProofError`] if proving fails.
 pub fn generate_proof(
```

Comment:
I'm still not sure I follow rationale of changing API here - the general API is `create_random_proof_with_reduction` which this specific function is hiding.

The RLNWitnessInput composite data structure seems OK to have in general, but I don't know why types are changed from Field to String etc.

Adding proving key here as requirement might be OK, depends a bit on how we imagine using this.

Reply:
WitnessInput before had string fields https://github.com/vacp2p/zerokit/blob/master/rln/src/main.rs#L26, I just kept this. But I agree that it is better to have them another time and manage serialization on the witness data structure.

Since builder takes as inputs BigInts when pushing fields, probably this would be the best type for witness fields, so that we can avoid some conversions on the proof generation.

I agree that the general API is `create_random_proof_with_reduction`. We can either:
- preprocess builder outside and have a RLN specific proc to fill the builder and use the general API `create_random_proof_with_reduction` which takes as input builder and proving key;
- have a RLN specific `generate_proof` wrapping `create_random_proof_with_reduction` that feeds the witness into the builder and computes a proof.

In both case we will have at least 1 RLN specific procedure, unclear to me which approach is better.